copy/copy.go

```diff
@@ -577,6 +577,29 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		}
 	}
 
+	destExistsCheck := true // TBD: enable configuration
```

Comment:
Maybe something in [containers.conf](https://github.com/containers/common/blob/master/pkg/config/containers.conf) eventually?

Reply:
No, this is not a system-wide decision; whether to do the check (= optimistically assume that the copy is a no-op) it is individual to the specific operation (is this a push of a just-built image that does not exist anywhere else vs. a daily sync of a rarely-changing large set of repositories).

So, for c/image this should be only a Go option with no user configuration, and I guess _most_ callers should leave this off. `skopeo sync` can start with hard-coding to on. Eventually we might encounter commands or cases where the right default is not obvious based on just the command, and where the difference is big enough, and that software can expose the option in the CLI or in individual config files.

Reply:
Thanks @TomSweeneyRedHat , @mtrmac, I have created an option `DestinationManifestCheck` in `copy.Options` (suggestions on naming completely welcome, there is likely a more descriptive name out there)

Reply:
Maybe something like `AssumeDestinationExists`? The API should describe the situation / assumption / desired behavior, not details of the implementation.

(Anyway, I’m afraid I still need to look into the details of correctness.)